app/addons/documents/mango/mango.api.js

```diff
@@ -84,6 +84,30 @@ export const fetchIndexes = (databaseName, params) => {
     });
 };
 
+// assume all databases being accessed are on the same
+// host / CouchDB version
+let supportsExecutionStatsCache = null;
+const supportsExecutionStats = (databaseName) => {
+  if (supportsExecutionStatsCache === null) {
+    return new FauxtonAPI.Promise((resolve) => {
+      mangoQuery(databaseName, {
+        selector: {
+          "_id": {"$gt": "a" }
+       },
+        execution_stats: true
+      }, {limit: 1})
+      .then(resp => {
+        supportsExecutionStatsCache = resp.status == 200;
+        resolve(supportsExecutionStatsCache);
+      });
+    });
+  } else {
+    return new FauxtonAPI.Promise((resolve) => {
```

Comment:
You can use [Promise.resolve](http://bluebirdjs.com/docs/api/promise.resolve.html) instead